hpx/util/tuple.hpp

```diff
@@ -35,7 +36,39 @@ namespace hpx { namespace util
 
     template <std::size_t I, typename T>
     struct tuple_element; // undefined
+}}
 
+// Adapt hpx::util::tuple to be useable with std::tuple
+namespace std
+{
+    template <std::size_t I, typename ...Ts>
+    class tuple_element<I, hpx::util::tuple<Ts...>>
+      : hpx::util::tuple_element<I, hpx::util::tuple<Ts...>>
+    {};
+
+    template <typename ...Ts>
+    struct tuple_size<hpx::util::tuple<Ts...>>
+      : hpx::util::tuple_size<hpx::util::tuple<Ts...>>
+    {};
+
+    template <std::size_t I, typename ...Ts>
+    typename hpx::util::tuple_element<I, hpx::util::tuple<Ts...>>::type&
+    get(hpx::util::tuple<Ts...>& tuple)
+    {
+        return hpx::util::tuple_element<I, hpx::util::tuple<Ts...>>::get(tuple);
+    }
+
+    template <std::size_t I, typename ...Ts>
+    typename hpx::util::tuple_element<I, hpx::util::tuple<Ts...>>::type const&
+    get(hpx::util::tuple<Ts...> const& tuple)
+    {
+        return hpx::util::tuple_element<I, hpx::util::tuple<Ts...>>::get(tuple);
+    }
+}
```

Comment:
Should we add overloads for rvalue tuples as well?

Reply:
Note that adding these overloads to namespace `std` is UB. If they are still wanted, then there should be 4 of them to match `std::` expected interface.

Reply:
The fact that it is UB can be neglected here. There should be no consequences for existing implementations or any implementation in the future.

Reply:
The overloads have been added.